benchbuild/experiment.py

```diff
@@ -73,26 +86,16 @@ class Experiment(metaclass=ExperimentRegistry):
             in the database scheme.
     """
 
-    NAME = None
-    SCHEMA = None
-
-    def __new__(cls, *args, **kwargs):
-        """Create a new experiment instance and set some defaults."""
-        del args, kwargs  # Temporarily unused
-        new_self = super(Experiment, cls).__new__(cls)
-        if cls.NAME is None:
-            raise AttributeError(
-                "{0} @ {1} does not define a NAME class attribute.".format(
-                    cls.__name__, cls.__module__))
-        return new_self
+    NAME: str = ''
+    SCHEMA: str = ''
 
-    name = attr.ib(
-        default=attr.Factory(lambda self: type(self).NAME, takes_self=True))
+    name: str = attr.ib(
+        default=attr.Factory(lambda self: type(self).NAME, takes_self=True),
+        validator=validators.attribute_not_empty('NAME'))
 
-    projects = \
-        attr.ib(default=attr.Factory(list))
+    projects: Dict[str, ProjectClass] = attr.ib(default=attr.Factory(list))
 
-    id = attr.ib()
+    id: uuid.UUID = attr.ib(validator=attr.validators.instance_of(uuid.UUID))
```

Comment:
C0103 (invalid-name): Class attribute name "id" doesn't conform to '([A-Za-z_][A-Za-z0-9_]{2,30}|(__.*__))$' pattern  <!-- Automatically posted by Lintly -->